wordpress-mobile/AztecEditor-iOS
# Issue/8045 aztec image links

This is the Aztec related work needed to implement [adding links to images in posts from its detail editor](https://github.com/wordpress-mobile/WordPress-iOS/issues/8045).

One small detail to note is that in the example app, the `linkURL` is automatically set to the same thing as the file URL after the image is added in EditorDemoController.

Doing it that way, rather than automatically setting it within ImageAttachment, leaves it up to the developer integrating AztecEditor into their app to decide whether or not they want to automatically link images to themselves.

To test:
1. run the example app and tap "Empty Editor Demo"
2. tap the text view to bring up the format bar
3. tap the "+" button in the lower left corner to add an image
4. once you've added the image, tap on it to show "Tap for options"
5. tap the image again, then tap "Media Details"
6. note that the link URL is shown (change it to something else if you want)
7. tap "Done"
8. bring up the format bar again and tap the `</>` button to show that the link URL is shown within the  HTML (with an `<a href></a>` containing the URL around the img tag)
9. if you want, edit the URL within the HTML
10. tap the `</>` button again to show that the HTML is properly converted back to an NSAttributedString... and that the image displays properly
11. bring up the image media details again, and observe that the link URL is still set (because it was set after converting back from HTML)
